WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderTagActions.java

```diff
@@ -102,51 +106,63 @@ public static boolean addTags(@NotNull final List<ReaderTag> tags,
                     ReaderTagType.FOLLOWED);
             newTags.add(newTag);
         }
-
-        com.wordpress.rest.RestRequest.Listener listener = jsonObject -> {
-            AppLog.i(T.READER, "add tag succeeded");
-            // the response will contain the list of the user's followed tags
-            ReaderTagList followedTags = parseFollowedTags(jsonObject);
-            ReaderTagTable.replaceFollowedTags(followedTags);
-            if (actionListener != null) {
-                ReaderActions.callActionListener(actionListener, true);
+        if (!isLoggedIn && AppPrefs.isReaderImprovementsPhase2Enabled()) {
+            // TODO: Revisit logic when ReaderTagRepository's getUserTags is implemented
+            if (!AppPrefs.getReaderRecommendedTagsDeletedForLoggedOutUser()) {
+                // Delete any previously saved recommended followed tags
+                ReaderTagTable.setRecommendedTags(new ReaderTagList());
```

Comment:
I believe the "replaceFollowedTags" method removes all the tags from the DB and inserts the provided tags. We might want to consider removing this line. Wdyt?

Reply:
What I meant was that we could just call `ReaderTagTable.replaceFollowedTags(newTags);` -> which removes the previous and inserts the new -> so we technically didn't need to call `ReaderTagTable.setRecommendedTags(new ReaderTagList());`. However, since you replaced it and we are removing also the posts for these tags, it's all good now;)! Thanks!